cpp/ps0004/ps0004_solv.cpp

```diff
@@ -4,30 +4,22 @@ using namespace std;
 
 int main()
 {
-    int x , counter;
+    // prompting user for number 1/1.5
+    float num;
+    cout << "\nPlease enter a number between 32 and 126:\t";
+    cin >> num;
 
-    cout << "Please enter the number of attempts \n";
-    cin>>counter;
-    for(int i=1;i<=counter;i++)
-    {
-      cout<<" \n Please enter a number between 32 and 126 \t ";
-      cin>>x;
-
-       if(x>=32 && x<=126)
-       {
-         cout<<"The character for "<< x <<" is "<< char(x) <<" . \n \n";
-       }
-
-       else if((x<0 || x<32 || x>126) && x!=0)
-        {
-         cout<<"Invalid : number is out of the range ! \n \n";
-        }
-
-       else if(x==0 || x== char(x))
-        {
-         cout<<"Invalid character/s or Zero value ! \n \n";
-        }
-
-    }
-return 0;
+    // checking invalidaty of user's input 2/4
+    // printing the number and the opposite printable character 2/2
+    if(num == 0)
+        cout << "Invalid character/s or Zero value!\n\n";
+    else if(num != (int)num)
+        cout << "Invalid: a floating point number!\n";
+    else if(num < 32 || num > 126)
+        cout << "Invalid : number is out of the range!\n\n";
+    else
+        cout << "The character for " << num << " is " << char(num) << " .\n\n";
+    
+    return 0;
 }
+// good work 👍 6/10

```

Comment:
- [x] program is in the correct path
- [ ] no syntax errors, bugs or warnings while compilation and execution
- [x] formatted code to be clear and readable
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
you can check the [problem solution](https://onlinegdb.com/9x6wLTlkK) for more